docker/stack.sh

```diff
@@ -0,0 +1,227 @@
+    -v "${TEMP_STORAGE_DIR}":/dispatcher/nfs \
+    "${DISPATCHER_REPO}":"${TAG}"
+}
+
+run_scheduler() {
+  local TAG="${1}"
+  local CONF_FILE="scheduler.conf"
+  docker run -dit \
+    --name "${SCHEDULER_CONTAINER}" \
+    -v "$(pwd)"/config/"${CONF_FILE}":/scheduler/"${CONF_FILE}" \
+    -v "$(pwd)"/config/log4j.properties:/scheduler/log4j.properties \
+    -v "$(pwd)"/resources/execution_script_tags.json:/dispatcher/resources/execution_script_tags.json \
+    "${SCHEDULER_REPO}":"${TAG}"
+}
+
+access_catalog() {
+  local CATALOG_IP=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' ${CATALOG_CONTAINER})
+  psql -h "${CATALOG_IP}" -p ${CATALOG_PORT} ${CATALOG_DB} ${CATALOG_USER}
+}
+
+access() {
+  local service="${1}"
+  case $service in
+    catalog)
+      access_catalog
+      ;;
+  esac
+}
+
+build() {
+  local service="${1}"
+  local tag="${2-latest}"
+  case $service in
+    archiver)
+      build_archiver ${tag}
+      ;;
+    catalog)
+      build_catalog ${tag}
+      ;;
+    dispatcher)
+      build_dispatcher ${tag}
+      ;;
+    scheduler)
+      build_scheduler ${tag}
+  esac
+}
+
+run() {
+  local service="${1}"
+  local tag="${2:-latest}"
+  case $service in
+    archiver)
+      run_archiver ${tag}
+      ;;
+    catalog)
+      run_catalog ${tag}
+      ;;
+    dispatcher)
+      run_dispatcher ${tag}
+      ;;
+    scheduler)
+      run_scheduler ${tag}
+  esac
+}
+
+restart() {
+  while [[ $# -ne 0 ]]
+  do
+    case $1 in
+      archiver) shift
+        tag=$(docker inspect -f '{{.Config.Image}}' ${ARCHIVER_CONTAINER} | cut -d':' -f2)
+        docker stop ${ARCHIVER_CONTAINER}
+        docker rm ${ARCHIVER_CONTAINER}
+        run archiver ${tag}
+        ;;
+      catalog) shift
+        tag=$(docker inspect -f '{{.Config.Image}}' ${CATALOG_CONTAINER} | cut -d':' -f2)
+        docker stop ${CATALOG_CONTAINER}
+        docker rm ${CATALOG_CONTAINER}
+        run catalog ${tag}
+        ;;
+      dispatcher) shift
+        tag=$(docker inspect -f '{{.Config.Image}}' ${DISPATCHER_CONTAINER} | cut -d':' -f2)
+        docker stop ${DISPATCHER_CONTAINER}
+        docker rm ${DISPATCHER_CONTAINER}
+        run dispatcher ${tag}
+        ;;
+      scheduler) shift
+        tag=$(docker inspect -f '{{.Config.Image}}' ${SCHEDULER_CONTAINER} | cut -d':' -f2)
+        docker stop ${SCHEDULER_CONTAINER}
+        docker rm ${SCHEDULER_CONTAINER}
+        run scheduler ${tag}
+        ;;
+      *) shift
+        ;;
+    esac
+  done
+}
+
+publish()
+{
+  local service="${1}"
+  local tag="${2:-latest}"
+  case $service in
+    archiver)
+      docker push "${ARCHIVER_REPO}":"${tag}"
+      ;;
+    catalog)
+      docker push "${CATALOG_REPO}":"${tag}"
+      ;;
+    dispatcher)
+      docker push "${DISPATCHER_REPO}":"${tag}"
+      ;;
+    scheduler)
+      docker push "${SCHEDULER_REPO}":"${tag}"
+      ;;
+  esac
+}
+
+define_params() {
+  case $1 in 
```

Comment:
just another suggestion. To make this script more unix-like, you could use getopts and when you call the script you should call like ./stack.sh -p access [other params]. If you have time, I strongly recommend that.

Reply:
I'm not against this suggestion ... but, you guys should keep in mind to keep things simple (and, beware of the difference between options and mandatory params)